This is the `numba` log:
```
Running phase: installCheckPhase
install: skipping file '/dev/fd/63', as it was replaced while being copied
/private/tmp/nix-build-python3.11-numba-0.60.0-unstable-2024-08-14.drv-0/tmp.MS5DfUHnaQ /private/tmp/nix-build-python3.11-numba-0.60.0-unstable-2024-08-14.drv-0/source
Parallel: 13. Serial: 0
Process SyncManager-1:
Traceback (most recent call last):
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/multiprocessing/managers.py", line 592, in _run_server
    server = cls._Server(registry, address, authkey, serializer)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/multiprocessing/managers.py", line 156, in __init__
    self.listener = Listener(address=address, backlog=128)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/multiprocessing/connection.py", line 464, in __init__
    self._listener = SocketListener(address, family, backlog)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/multiprocessing/connection.py", line 608, in __init__
    self._socket.bind(address)
OSError: AF_UNIX path too long
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/nix/store/bwlm5zrzab23wjf5qp3iapyrm1h8xh5r-python3.11-numba-0.60.0-unstable-2024-08-14/lib/python3.11/site-packages/numba/runtests.py", line 9, in <module>
    sys.exit(0 if _main(sys.argv) else 1)
                  ^^^^^^^^^^^^^^^
  File "/nix/store/bwlm5zrzab23wjf5qp3iapyrm1h8xh5r-python3.11-numba-0.60.0-unstable-2024-08-14/lib/python3.11/site-packages/numba/testing/_runtests.py", line 25, in _main
    return run_tests(argv, defaultTest='numba.tests',
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/bwlm5zrzab23wjf5qp3iapyrm1h8xh5r-python3.11-numba-0.60.0-unstable-2024-08-14/lib/python3.11/site-packages/numba/testing/__init__.py", line 54, in run_tests
    prog = NumbaTestProgram(argv=argv,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/bwlm5zrzab23wjf5qp3iapyrm1h8xh5r-python3.11-numba-0.60.0-unstable-2024-08-14/lib/python3.11/site-packages/numba/testing/main.py", line 207, in __init__
    super(NumbaTestProgram, self).__init__(*args, **kwargs)
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/unittest/main.py", line 102, in __init__
    self.runTests()
  File "/nix/store/bwlm5zrzab23wjf5qp3iapyrm1h8xh5r-python3.11-numba-0.60.0-unstable-2024-08-14/lib/python3.11/site-packages/numba/testing/main.py", line 386, in runTests
    run_tests_real()
  File "/nix/store/bwlm5zrzab23wjf5qp3iapyrm1h8xh5r-python3.11-numba-0.60.0-unstable-2024-08-14/lib/python3.11/site-packages/numba/testing/main.py", line 371, in run_tests_real
    super(NumbaTestProgram, self).runTests()
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/unittest/main.py", line 274, in runTests
    self.result = testRunner.run(self.test)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/bwlm5zrzab23wjf5qp3iapyrm1h8xh5r-python3.11-numba-0.60.0-unstable-2024-08-14/lib/python3.11/site-packages/numba/testing/main.py", line 976, in run
    result = super(ParallelTestRunner, self).run(self._run_inner)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/unittest/runner.py", line 217, in run
    test(result)
  File "/nix/store/bwlm5zrzab23wjf5qp3iapyrm1h8xh5r-python3.11-numba-0.60.0-unstable-2024-08-14/lib/python3.11/site-packages/numba/testing/main.py", line 894, in _run_inner
    with multiprocessing.Manager() as manager:
         ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/multiprocessing/context.py", line 57, in Manager
    m.start()
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/multiprocessing/managers.py", line 567, in start
    self._address = reader.recv()
                    ^^^^^^^^^^^^^
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/multiprocessing/connection.py", line 250, in recv
    buf = self._recv_bytes()
          ^^^^^^^^^^^^^^^^^^
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/multiprocessing/connection.py", line 430, in _recv_bytes
    buf = self._recv(4)
          ^^^^^^^^^^^^^
  File "/nix/store/kcm5282kss1mnil624lccr7y8wqg75vk-python3-3.11.9/lib/python3.11/multiprocessing/connection.py", line 399, in _recv
    raise EOFError
EOFError
```

--- a/pkgs/development/python-modules/llvmlite/default.nix
+++ b/pkgs/development/python-modules/llvmlite/default.nix
@@ -34,7 +34,8 @@ buildPythonPackage rec {
   ];
 
   postPatch = ''
-    substituteInPlace llvmlite/tests/test_binding.py --replace "test_linux" "nope"
+    substituteInPlace llvmlite/tests/test_binding.py \
+      --replace-fail "test_linux" "nope"
   '';
 
   # Set directory containing llvm-config binary

--- a/pkgs/development/python-modules/llvmlite/default.nix
+++ b/pkgs/development/python-modules/llvmlite/default.nix
@@ -11,7 +11,7 @@
   setuptools,
 
   # tests
-  python,
+  pytestCheckHook,
 }:
 
 buildPythonPackage rec {
@@ -46,10 +46,12 @@ buildPythonPackage rec {
     export LLVM_CONFIG=${llvm.dev}/bin/llvm-config
   '';
 
-  checkPhase = ''
-    runHook preCheck
-    ${python.executable} runtests.py
-    runHook postCheck
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+  # https://github.com/NixOS/nixpkgs/issues/255262
+  preCheck = ''
+    cd $out
   '';
 
   __impureHostDeps = lib.optionals stdenv.isDarwin [ "/usr/lib/libm.dylib" ];

--- a/pkgs/development/python-modules/numba/default.nix
+++ b/pkgs/development/python-modules/numba/default.nix
@@ -7,7 +7,6 @@
   python,
   buildPythonPackage,
   setuptools,
-  numpy,
   numpy_2,
   llvmlite,
   libcxx,
@@ -68,7 +67,6 @@ buildPythonPackage rec {
 
   build-system = [
     setuptools
-    numpy_2
   ];
 
   nativeBuildInputs = lib.optionals cudaSupport [
@@ -76,10 +74,12 @@ buildPythonPackage rec {
     cudaPackages.cuda_nvcc
   ];
 
-  buildInputs = lib.optionals cudaSupport [ cudaPackages.cuda_cudart ];
+  buildInputs = [
+    # Not propagating it, because it numba can work with either numpy_2 or numpy_1
+    numpy_2
+  ] ++ lib.optionals cudaSupport [ cudaPackages.cuda_cudart ];
 
   dependencies = [
-    numpy
     llvmlite
     setuptools
   ] ++ lib.optionals (pythonOlder "3.9") [ importlib-metadata ];

